src/StaticLibrary.h

```diff
@@ -22,6 +22,19 @@ namespace Internal {
 EXPORT void create_ar_file(const std::vector<std::string> &src_files, 
                            const std::string &dst_file, bool deterministic = true);
 
+/**
+ * Given a list of "files" (really, names and data), create an ar file.
+ * Note that 'data' is a string, but may hold binary data; do *not*
```

Comment:
That's what I started using at first, but it ended up adding extra boilerplate overhead in terms of code; it's totally legit to use std::string this way (strings can contain binary data just fine as long as you don't use c_str()). If you think this is a bad practice then I'll revise to use std::vector<uint8_t> instead, it will just expand code size a bit.